src/spatial-index.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  DEFAULT_AGGREGATION_RES_LEVEL_H3,
+  DEFAULT_AGGREGATION_RES_LEVEL_QUADBIN,
+} from './constants-internal';
+import {ModelSource} from './models/model';
+import {AggregationOptions} from './sources/types';
+
+const DEFAULT_TILE_SIZE = 512;
+const QUADBIN_ZOOM_MAX_OFFSET = 4;
+
+export function getSpatialFiltersResolution({
```

Comment:
Does similar logic exist in C4R now, and could you share a link to it for comparison? 

I'm not sure about using `tileSize` without also referring to `tileResolution`, that seems a bit strange to me, but maybe this is just how it already works?

Reply:
Yes this is almost copied verbatim from C4R. See here https://github.com/CartoDB/carto-react/blob/master/packages/react-widgets/src/models/spatialFiltersResolution.js#L43

Reply:
I see, thanks! I think at least for the `aggregationResLevel` we should make sure the user doesn't need to specify that twice (once for the source, and again for getSpatialFiltersResolution). But there's more than one way to do this, so I think it will depend on the question above.

Reply:
Okay I think with the last implementation we can consider this conversation close, since right now the user does not have to pass any source props to the widget model call

Reply:
Yes, I think so! I still want to look at https://github.com/CartoDB/carto-api-client/issues/39 later but that shouldn't block anything here.